root/etc/cont-init.d/45-plex-hw-transcode-and-connected-tuner

```diff
@@ -5,28 +5,23 @@ if [ ! -e /dev/dri ] && [ ! -e /dev/dvb ]; then
 	exit 0
 fi
 
-# Get the group ID for the dri or dvb devices.
-if [ -e /dev/dri ]; then
-  DEVICE_GID=$(stat -c '%g' /dev/dri/* | grep -v '^0$' | head -n 1)
-else
-  DEVICE_GID=$(stat -c '%g' /dev/dvb/adapter*/* | grep -v '^0$' | head -n 1)
-fi
-
-# Get the group ID for the video group
-VIDEO_GID=$(getent group video | awk -F: '{print $3}')
-
-# If the video group's ID matches the group ID of the device, exit as permissions are already setup.
-if [ "${DEVICE_GID}" == "${VIDEO_GID}" ]; then
-	exit 0
-fi
+FILES=$(find /dev/dri /dev/dvb -type c -print 2>/dev/null)
 
-# Get the current group name for the device's group ID
-CURRENT_GROUP=$(getent group ${DEVICE_GID} | awk -F: '{print $1}')
+for i in $FILES
+do
+# Get the group ID for the dri or dvb device.
+    DEVICE_GID=$(stat -c '%g' $i)
+# Get the group name (if it exists)
+    CURRENT_GROUP=$(getent group "${DEVICE_GID}" | awk -F: '{print $1}')
+# If it doesn't exist, create a new group name and assign it to the device GID
+    if [ -z "${CURRENT_GROUP}" ]; then
+	CURRENT_GROUP=video$(head /dev/urandom | tr -dc 'a-zA-Z0-9' | head -c4)
```

Comment:
Rather than random, use a counter so there is always uniqueness:
```
# Outside of the for loop
GROUP_COUNT=1
…
CURRENT_GROUP=video${GROUP_COUNT}
GROUP_COUNT=$(($GROUP_COUNT + 1))
…
```